components/create_comment/create_comment.tsx

```diff
@@ -551,11 +553,20 @@ class CreateComment extends React.PureComponent<Props, State> {
     }
 
     handlePostError = (postError: React.ReactNode) => {
-        this.setState({postError});
+        const state = {postError} as State;
+
+        if (postError) {
+            state.deferringPostSubmit = false;
+        }
+
+        this.setState(state);
     }
 
-    handleSubmit = async (e: React.FormEvent | React.MouseEvent) => {
-        e.preventDefault();
+    handleSubmit = async (e?: React.FormEvent | React.MouseEvent) => {
+        if (e) {
+            e.preventDefault();
+        }
```

Comment:
`e` should always be there when handed down to a React component

Reply:
@michelengelen I was following suit with how `doSubmit` (which is called by this `handleSubmit` function) handles receiving a null event. I am manually calling `handleSubmit` in this PR to trigger a submit programmatically, and I don't have access to an event at that time.

I'm actually not sure if the new code should be calling `handleSubmit` or `doSubmit` though. `handleSubmit` seems to be doing a lot of checks that seem necessary. Though `handleNotifyAllConfirmation` calls `doSubmit` without calling `handleSubmit`, so I'm not entirely sure which function should be called in the deferred post case. Do you mind taking a look at `handleSubmit` and `doSubmit` and let me know your thoughts?

Reply:
The exact same thing is happening in `create_post.tsx` as well:

`handleNotifyAllConfirmation` calls `doSubmit` without calling `handleSubmit`

https://github.com/mattermost/mattermost-webapp/blob/c7724d09dcb0f4e724530bace5d623ef6c28c162/components/create_post/create_post.tsx#L628-L631